src/util/message_util.go

```diff
@@ -58,3 +59,37 @@ func ProduceMessageToChannel(channel chan common.DetailedLogsBatch, currentBatch
 		Entries: currentBatch,
 	}}
 }
+
+// CloudTrailRecords represents a list of CloudTrail records.
+type CloudTrailRecords struct {
+	Records []map[string]interface{} `json:"Records"`
+}
+
+// ParseCloudTrailEvents parses a CloudTrail message and returns a list of log records as strings.
+func ParseCloudTrailEvents(message string) ([]string, error) {
+	var cloudTrailRecords CloudTrailRecords
+	err := json.Unmarshal([]byte(message), &cloudTrailRecords)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Serialize each record into a JSON string.
+	var records []string
+	for _, record := range cloudTrailRecords.Records {
+		if record["eventTime"] != nil {
+			parsedTime, err := time.Parse(time.RFC3339, record["eventTime"].(string))
+			if err == nil {
+				record["timestamp"] = parsedTime.Unix()
```

Comment:
setting it here worked? Log appearing correctly in NR One?

Reply:
yes